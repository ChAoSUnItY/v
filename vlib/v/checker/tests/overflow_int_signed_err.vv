i8s := [
	i8(0xff), // converted to -1
	i8(128), // converted to -128
	i8(-129), // converted to +127
	i8(-0xff), // converted to +1
]

i16s := [
	i16(0xffff), // converted to -1
	i16(32768), // converted to -32768
	i16(-32769), // converted to +32767
	i16(-0xffff), // converted to +1
]

ints := [
	int(0xffffffff), // converted to -1
	int(2147483648), // converted to -2147483648
	int(-2147483649), // converted to +2147483647
	int(-0xffffffff), // converted to +1
]

i64s := [
	i64(0xffffffffffffffff), // converted to -1
	i64(9223372036854775808), // converted to -9223372036854775808
	i64(-9223372036854775809), // converted to +9223372036854775807
	i64(-0xffffffffffffffff), // converted to +1
]

println('i8s: ${i8s}')
println('i16s: ${i16s}')
println('ints: ${ints}')
println('i64s: ${i64s}')
