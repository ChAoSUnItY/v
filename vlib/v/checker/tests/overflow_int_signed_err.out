vlib/v/checker/tests/overflow_int_signed_err.vv:2:2: error: value `0xff overflows `i8`
    1 | i8s := [
    2 |     i8(0xff)  // converted to -1
      |     ~~~~~~~~
    3 |     i8(128)   // converted to -128
    4 |     i8(-129)  // converted to +127
vlib/v/checker/tests/overflow_int_signed_err.vv:3:2: error: value `128 overflows `i8`
    1 | i8s := [
    2 |     i8(0xff)  // converted to -1
    3 |     i8(128)   // converted to -128
      |     ~~~~~~~
    4 |     i8(-129)  // converted to +127
    5 |     i8(-0xff) // converted to +1
vlib/v/checker/tests/overflow_int_signed_err.vv:9:2: error: value `0xffff overflows `i16`
    7 |
    8 | i16s := [
    9 |     i16(0xffff)    // converted to -1
      |     ~~~~~~~~~~~
   10 |     i16(32768)     // converted to -32768
   11 |     i16(-32769)    // converted to +32767
vlib/v/checker/tests/overflow_int_signed_err.vv:10:2: error: value `32768 overflows `i16`
    8 | i16s := [
    9 |     i16(0xffff)    // converted to -1
   10 |     i16(32768)     // converted to -32768
      |     ~~~~~~~~~~
   11 |     i16(-32769)    // converted to +32767
   12 |     i16(-0xffff)   // converted to +1
vlib/v/checker/tests/overflow_int_signed_err.vv:16:2: error: value `0xffffffff overflows `int`
   14 |
   15 | ints := [
   16 |     int(0xffffffff)  // converted to -1
      |     ~~~~~~~~~~~~~~~
   17 |     int(2147483648)  // converted to -2147483648
   18 |     int(-2147483649) // converted to +2147483647
vlib/v/checker/tests/overflow_int_signed_err.vv:17:2: error: value `2147483648 overflows `int`
   15 | ints := [
   16 |     int(0xffffffff)  // converted to -1
   17 |     int(2147483648)  // converted to -2147483648
      |     ~~~~~~~~~~~~~~~
   18 |     int(-2147483649) // converted to +2147483647
   19 |     int(-0xffffffff) // converted to +1
vlib/v/checker/tests/overflow_int_signed_err.vv:23:2: error: value `0xffffffffffffffff overflows `i64`
   21 |
   22 | i64s := [
   23 |     i64(0xffffffffffffffff)  // converted to -1
      |     ~~~~~~~~~~~~~~~~~~~~~~~
   24 |     i64(9223372036854775808) // converted to -9223372036854775808
   25 |     i64(-9223372036854775809) // converted to +9223372036854775807
vlib/v/checker/tests/overflow_int_signed_err.vv:24:2: error: value `9223372036854775808 overflows `i64`
   22 | i64s := [
   23 |     i64(0xffffffffffffffff)  // converted to -1
   24 |     i64(9223372036854775808) // converted to -9223372036854775808
      |     ~~~~~~~~~~~~~~~~~~~~~~~~
   25 |     i64(-9223372036854775809) // converted to +9223372036854775807
   26 |     i64(-0xffffffffffffffff)  // converted to +1